Metalama.Framework.Engine/Validation/ValidationRunner.cs

```diff
@@ -78,32 +82,41 @@ public void RunDeclarationValidators( CompilationModel compilation, CompilationM
         }
     }
 
-    public async Task RunDeclarationValidatorsAsync(
+    public async Task<ImmutableArray<DeclarationValidatorInstance>> RunDeclarationValidatorsAsync(
         CompilationModel initialCompilation,
         CompilationModel finalCompilation,
         UserDiagnosticSink diagnosticAdder,
         CancellationToken cancellationToken )
     {
-        var task1 = this.RunDeclarationValidatorsAsync( initialCompilation, CompilationModelVersion.Initial, diagnosticAdder, cancellationToken );
-        var task2 = this.RunDeclarationValidatorsAsync( finalCompilation, CompilationModelVersion.Final, diagnosticAdder, cancellationToken );
+        var initialCompilationValidationTask = this.RunDeclarationValidatorsAsync( initialCompilation, CompilationModelVersion.Initial, diagnosticAdder, cancellationToken );
+        var finalCompilationValidationTask = this.RunDeclarationValidatorsAsync( finalCompilation, CompilationModelVersion.Final, diagnosticAdder, cancellationToken );
 
-        await Task.WhenAll( task1, task2 );
+        await Task.WhenAll( initialCompilationValidationTask, finalCompilationValidationTask );
+        
+        var initialCompilationValidators = await initialCompilationValidationTask;
+        var finalCompilationValidators = await finalCompilationValidationTask;
+        var allDeclarationValidators = initialCompilationValidators.Union( finalCompilationValidators ).ToImmutableArray();
+
+        return allDeclarationValidators;
```

Comment:
Collecting all declaration validators is useless. A `bool` return value `HasDeclarationValidator` is enough.